atom/browser/api/atom_api_session.cc

```diff
@@ -232,6 +241,12 @@ void Session::ClearStorageData(mate::Arguments* args) {
       base::Time(), base::Time::Max(), callback);
 }
 
+void Session::SetProxy(const std::string& proxy) {
+  auto getter = browser_context_->GetRequestContext();
```

Comment:
yeah i was confused about its use, as it states to be called in UI thread [here](https://code.google.com/p/chromium/codesearch#chromium/src/content/public/browser/browser_context.h&sq=package:chromium&rcl=1437050479&l=136).